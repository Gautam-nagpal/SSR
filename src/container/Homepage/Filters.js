import React from "react";
import { countArray } from "../../utils";
import { Subfilters } from "../../components"


function Filters(props) {
    const { onChangeFilters, selectedFilters } = props;
    let yearArr = countArray(2006, 15)
    return (
        <div className="filters-section">
            <h4 className="title-heading"> Filters</h4>
            <div>
                <Subfilters
                    onChangeFilters={onChangeFilters}
                    type="launch_year"
                    title="Launch Year"
                    itemsArr={yearArr}
                    selectedFilters={selectedFilters}
                />
            </div>

            <div>
                <Subfilters
                    onChangeFilters={onChangeFilters}
                    type="launch_success"
                    title="Successful Launch"
                    itemsArr={[true, false]}
                    selectedFilters={selectedFilters}
                />
            </div>

            <div>
                <Subfilters
                    onChangeFilters={onChangeFilters}
                    type="land_success"
                    title="Successful Land"
                    itemsArr={[true, false]}
                    selectedFilters={selectedFilters}
                />
            </div>
        </div>
    )
}

export default Filters;


