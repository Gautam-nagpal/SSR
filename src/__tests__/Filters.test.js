import React from 'react';
import { mount, shallow } from 'enzyme';
import Filters from '../container/Homepage/Filters';


let selectedFilters = {}

const onChangeFilters = () => {

}

describe('Renders Filters Component', () => {
    let wrapper = shallow(<Filters onChangeFilters={onChangeFilters} selectedFilters={selectedFilters || {}} />);

    it('should render correctly', () => {
        expect(wrapper).toMatchSnapshot();
    })

    it('should render div', () => {
        expect(wrapper.find(".filters-section").length).toBe(1);
    });


    it('should render div', () => {
        expect(wrapper.find(".filters-section").length).toBe(1);
    });
    
});
