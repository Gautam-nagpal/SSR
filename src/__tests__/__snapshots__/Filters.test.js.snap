// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Filters Component should render correctly 1`] = `ShallowWrapper {}`;
